src/ILCompiler/src/Program.cs

```diff
@@ -238,6 +239,9 @@ private int Run(string[] args)
             if (_outputFilePath == null)
                 throw new CommandLineException("Output filename must be specified (/out <file>)");
 
+            // Create the output directory if it doesn't exist yet
+            Directory.CreateDirectory(Path.GetDirectoryName(_outputFilePath));
```

Comment:
None of the compilers I work with do this:

```sh
michals@michals-dev:~$ gcc test.c -o ~/createme/test
/usr/bin/ld: cannot open output file /home/michals/createme/test: No such file or directory
collect2: error: ld returned 1 exit status
```

```
D:\>csc test.cs -out:d:\createme\test.exe
error CS2012: Cannot open 'd:\createme\test.exe' for writing -- 'Could not find a part of the path 'd:\createme\test.exe'.'
```

Creating directories is another of those things that the build orchestrator should do. It's not compiler's job. I know it's what .NET Native's ILC would do, but I tend to take .NET Native ILC as a negative template (same way as Linus took CVS as a [negative template](https://www.google.com/search?q=WWCVSND) when designing Git). .NET Native is half-compiler, half-msbuild. Let's just build a compiler here.

(Plus having an extra spot that can fail with an IOException is annoying.)